### ISSUE TITLE
Will CoreML Conversion Support be Available for YOLOv10 Custom Models?

### ISSUE THREAD

**Issue by adkbbx** [ultralytics/ultralytics#13342]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I am planning to use the YOLOv10 Ultralytics integrated custom model on my iOS app after the official integration is completed on the Ultralytics package. While I wait for the update, I was wondering if CoreML conversion support will also be available, similar to the YOLOv8 custom model conversion. This feature would be really useful for deploying models on Apple devices.

Thank you for your reply in advance!

**Comment by glenn-jocher** [comment]
@adkbbx hello,

Thank you for reaching out with your question! As of now, CoreML conversion support is available for YOLOv8 models, and we aim to provide similar support for YOLOv10 models once they are officially integrated into the Ultralytics package.

We understand the importance of deploying models on Apple devices and are working towards ensuring that our newer models continue to support CoreML conversions. Keep an eye on our updates for the latest enhancements and feature rollouts.

Thank you for your patience and for choosing Ultralytics!

**Comment by Oil3** [comment]
@adkbbx hi,
in July I had tried and could have a compilable model, objects were accurately detected however the coordinates outputted were wrong (like allways the same).
I tried again, I'm not quite sure what change was different but now it seems it's all good.
The CoreML "n" version is five times faster than the PyTorch "n" version.
The "X" version is more than ten times faster.
Compared to V8, the "n" version is half a ms slower, but the "x" version is a couple ms faster, both in CoreML.
![image](https://github.com/user-attachments/assets/aa164669-5a34-493e-a030-0626c42987f4)

I uploaded v10n and v10s here https://github.com/Oil3/yolov10-coreML
